docs/shadertrap_language.md

```diff
@@ -0,0 +1,338 @@
+
+```
+# Sets the vec2 uniform at location 1 in my_prog to (1.0, 2.0)
+SET_UNIFORM LOCATION 1 TYPE vec2 VALUES 1.0 2.0 PROGRAM my_prog
+
+# Also sets the vec2 uniform at location 1 in my_prog to (1.0, 2.0)
+SET_UNIFORM PROGRAM my_prog VALUES 1.0 2.0 TYPE vec2 LOCATION 1
+```
+
+A number of complete ShaderTrap programs are provided in the `examples` directory.
+
+A brief description of each ShaderTrap command now follows. The commands are presented in alphabetical order.
+
+### ASSERT_EQUAL
+
+This command has two forms.
+
+```
+ASSERT_EQUAL BUFFERS buffer_1 buffer_2
+```
+
+Checks whether `buffer_1` and `buffer_2`, which must be buffers produced by `CREATE_BUFFER`, contain equal contents. Performs a byte-level comparison of the buffers. Yields an error if the buffers have different sizes, or if they have the same size but differ at any single byte.
+
+```
+ASSERT_EQUAL RENDERBUFFERS renderbuffer_1 renderbuffer_2
+```
+
+Checks whether `renderbuffer_1` and `renderbuffer_2`, which must be renderbuffers produced by `CREATE_RENDERBUFFER`, contain equal contents. Performs a pixel-by-pixel comparison of the renderbuffers. Yields an error if the renderbuffers have different widths or different heights, or if they have the same dimensions but differ at any single pixel.
+
+### ASSERT_PIXELS
+
+```
+ASSERT_PIXELS RENDERBUFFER renderbuffer RECTANGLE x y w h EXPECTED r g b a
+```
+
+Checks whether every pixel in a particular rectangular region of a renderbuffer has a specific value.
+- `renderbuffer` must be a renderbuffer produced by `CREATE_RENDERBUFFER`
+- `x`, `y`, `w`, `h` defines a rectangle with top-left coordinate (`x`, `y`), width `w` and height `h` that is required to be within the bounds of `renderbuffer`
+- `r`, `g`, `b` and `a` are integer values in the range [0, 255] that define the expected value for every pixel in the rectangular region
+
+### ASSERT_SIMILAR_EMD_HISTOGRAM
+
+```
+ASSERT_SIMILAR_EMD_HISTOGRAM RENDERBUFFERS renderbuffer_1 renderbuffer_2 TOLERANCE value
+```
+
+Checks whether two renderbuffers are similar according to the [Earth Mover's Distance](https://en.wikipedia.org/wiki/Earth_mover%27s_distance) metric.
+
+- `renderbuffer_1` and `renderbuffer_2` must be renderbuffers produced by `CREATE_RENDERBUFFER`
+- `value` must be a floating-point value specifying the tolerance to be used for the comparison. It is a unit-less number.
```

Comment:
...although it is a number in the range 0-1.

Reply:
Clarified, and opened #29 so that it can be checked.